akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -232,8 +232,8 @@ private[remote] class Association(
         _outboundControlIngress match {
           case OptionVal.Some(o) => o
           case OptionVal.None =>
-            if (transport.isShutdown) throw ShuttingDown
-            else throw new IllegalStateException("outboundControlIngress not initialized yet")
+            if (transport.isShutdown || isRemovedAfterQuarantined()) throw ShuttingDown
```

Comment:
looks good, the write to the queues is before the write to the volatile outboundCOntrolIngress field